### ISSUE TITLE
Overall Summary

### ISSUE THREAD

**Issue by sjlagasse** [tommygebru/PWJ_lodash#9]
### Criteria 1: Functionality
* _Score Level: 2/4
* _Comment(s): I had a hard time trying to get the tests to run. You had your own tests built in that were causing your functions to conflict with the actual tests. After moving some things around I was able to get half of them to pass. I also had to move the code from the `exercise` file to the `_` file.

### Criteria 2: Code Style - Formatting
* _Score Level: 1/4
* _Comment(s): I like that you added comments to your code. However, there was a ton of code that was commented out and not used. Also, the majority of the file didn't have any indentation.

### Criteria 3: Code Style - JavaScript Techniques
* _Score Level: 2/4
* _Comment(s): I think you still have a bit of work to do on your JavaScript fundamentals. You used `let` throughout the file and didn't use `const`. In your first few functions it seemed like you over engineered the solution. For instance, you used the `arguments` keyword and turned your parameters into objects. Another example is on line 117 you do `new Object();`. 

### Overall Score: 5/12

You're on the right track but I think you should review some of the lessons. You did a good job with loops and adding comments to all of your code. However, your code was in the wrong file and the tests didn't run. Make sure that you follow the instructions on how to test the file and what file to put your code in. I would also review when to use `const` as it should always be used over `let` when possible. Lastly, a lot of your functions were hard to follow. There was a lot of extra code that needed to be removed and things were over engineered. Make sure to use the parameters directly instead of creating and an object out of them then working with them. Make sure to keep these things in mind for your next JavaScript project.

**Comment by tommygebru** [comment]
Hey @sjlagasse thanks for reviewing my capstone, I was curious for this type of project (js) would it make sense to fork and share comments/feedback as a link or maybe a github gist might work too?

I am curious to learn more about the methods which failed, and which were really good and if there were alternative solutions as well

**Comment by sjlagasse** [comment]
Not sure what you mean. If you want to work through the tests yourself copy and paste your code in `exercise.js` into `_.js`, and run `node test-all.js` when you're inside of the test folder in the command line.

**Comment by tommygebru** [comment]
Sure ok I will look into that, this capstone had multiple options of submission and completion so I wasnt sure which was recommended, I was hoping for a deeper dive into each exercise and going from there 👍 

Ok thanks again Steven 👋
